### ISSUE TITLE
refresh api should be retired

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#12775]
With the availability of load-and-stream uploading sstables using nodetool refresh is deemed unsafe and so the refresh api should be first deprecated (by documentation and a warning at runtime) and eventually be removed from the code base.

**Comment by bhalevy** [comment]
Cc @avikivity @tzach @raphaelsc @asias

**Comment by tzach** [comment]
Maybe there is a use for ignoring non relevant ranges, not streaming them.
Even if this is the case, we can make load and stream teh default.
Note this (very) long pending issue https://github.com/scylladb/scylla-tools-java/issues/282

**Comment by asias** [comment]
Another reason to use load and stream is that there is no need to run cleanup after nodetool refresh

**Comment by bhalevy** [comment]
> Maybe there is a use for ignoring non relevant ranges, not streaming them. Even if this is the case, we can make load and stream teh default. Note this (very) long pending issue [scylladb/scylla-tools-java#282](https://github.com/scylladb/scylla-tools-java/issues/282)

FWIW, this is not really relevant with object storage in mind.
In this case restoring from backup would load'n'stream directly from objects without having to copy the sstables to local storage on the nodes and streaming them from their.

With that in mind, we can arbitrarily select nodes/shards to perform the load'n'stream, say in round robin, in they'd act just like coordinators, where the local token ownership is irrelevant to the load'n'stream operation.

Bottom line is that IMO in the long run we won't need to consider primary ranges in load_and_stream at all.

**Comment by asias** [comment]
> > Maybe there is a use for ignoring non relevant ranges, not streaming them. Even if this is the case, we can make load and stream teh default. Note this (very) long pending issue [scylladb/scylla-tools-java#282](https://github.com/scylladb/scylla-tools-java/issues/282)
> 
> FWIW, this is not really relevant with object storage in mind. In this case restoring from backup would load'n'stream directly from objects without having to copy the sstables to local storage on the nodes and streaming them from there.
> 
> With that in mind, we can arbitrarily select nodes/shards to perform the load'n'stream, say in round robin, in they'd act just like coordinators, where the local token ownership is irrelevant to the load'n'stream operation.
> 
> Bottom line is that IMO in the long run we won't need to consider primary ranges in load_and_stream at all.

The mentioned primary range option is about the destination replica node the load and stream should send to. It is not aboujt that load and stream can only process data belongs to the current node.  With load and stream, any node can ingress any data.

**Comment by kbr-scylla** [comment]
> load-and-stream

Please define this term

**Comment by raphaelsc** [comment]
> > load-and-stream
> 
> Please define this term

df3ef800c20c60d7929ffa600649793aa6c73064

**Comment by denesb** [comment]
I don't think we shoud retire the plain refresh. `--load-and-stream` is not universally superior to plain refresh, I can think of several different scenarios:
* cluster is in bad state, so streaming will keep failing
* we want to force-upload data to a node, without ownership checks
* optimization: sstable is known to be fully owned by the node and we don't want the overhead of load-and-stream

**Comment by bhalevy** [comment]
> I don't think we shoud retire the plain refresh. `--load-and-stream` is not universally superior to plain refresh, I can think of several different scenarios:
> 
> * cluster is in bad state, so streaming will keep failing
> * we want to force-upload data to a node, without ownership checks
> * optimization: sstable is known to be fully owned by the node and we don't want the overhead of load-and-stream

yes, and er continue to use it for 1:1 restore
